hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java

```diff
@@ -1374,28 +1394,31 @@ private void addKeyPrefixInfoToResultList(String keyPrefix,
         return;
       }
 
-      // TODO: HDDS-4859 will fix the case where startKey not started with
-      //  keyPrefix.
-
-      OzoneFileStatus status = proxy.getOzoneFileStatus(volumeName, name,
-          keyPrefix);
+      OzoneFileStatus status;
+      try {
+        status = proxy.getOzoneFileStatus(volumeName, name,
+            keyPrefix);
+      } catch (OMException ome) {
+        // ignore and return empty result list
```

Comment:
should ignore ome only in case of filenotfound exception

Reply:
yes, thats a good point. Sure will fix this.